lnwire/update_add_htlc_test.go

```diff
@@ -0,0 +1,216 @@
+package lnwire
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/tlv"
+	"github.com/stretchr/testify/require"
+)
+
+// testCase is a test case for the UpdateAddHTLC message.
+type testCase struct {
+	// Msg is the message to be encoded and decoded.
+	Msg UpdateAddHTLC
+
+	// ExpectEncodeError is a flag that indicates whether we expect the
+	// encoding of the message to fail.
+	ExpectEncodeError bool
+}
+
+// generateTestCases generates a set of UpdateAddHTLC message test cases.
+func generateTestCases(t *testing.T) []testCase {
```

Comment:
Probably due to the commit author not being familiar with either. Would count myself into that camp as well. Will invest some time into learning one of them (I assume `rapid` would be the more obvious choice since it probably is more feature rich?).

Reply:
Yeah the consensus is that new tests should move towards rapid. We have plenty of examples using quick right now to get an intuition for test design. However, quick has been abandoned and rapid has better support. CC @Roasbeef